### ISSUE TITLE
HttpLib2 Missing Header Error When Timezone Is Incorrect

### ISSUE THREAD

**Issue by pkenjora** [googleapis/google-api-python-client#1006]
#### Environment details

  - OS type and version: OSX
  - Python version: `Python 3.7.4`
  - pip version: `pip 20.2.2`
  - `google-api-python-client` version: `Version: 1.10.0`

#### Steps to reproduce

**When calling this API endpoint:**

https://developers.google.com/bid-manager/v1.1/queries/createquery

**With this field set to ( should be "America/Los_Angeles" ):**

timezoneCode:"Los Angeles/America"

**The following error shows up consistently at the wrong layer:**

httplib2.RedirectMissingLocation: Redirected but response is missing a location: header

#### Code example

```python
See above.
```

This is a bug that took hours to track down because the error is manifesting in httplib.  Which is the wrong layer to catch this.

Lookin online this crops up every few months and burns a bunch of time.  Hopefully having this be reproducible will help.

PS: Look me up at Google if you need to reproduce.

**Comment by busunkim96** [comment]
@pkenjora Are you using this in internal code? The internal copy of this library is behind the external one. What version of `httplib2` are you using?

The error looks like #891 and is probably from the API returning a status code that httplib2 treats as a redirect by default.

**Comment by shadow-light** [comment]
I'm getting this as well for `bq load`

```
Google Cloud SDK 306.0.0
alpha 2020.08.14
beta 2020.08.14
bq 2.0.59
core 2020.08.14
gsutil 4.52
kubectl 1.15.11
```

**Comment by shadow-light** [comment]
I don't know what the underlying issue is, but error is `BigQuery error in load operation: Could not connect with BigQuery server due to: RedirectMissingLocation('Redirected but the response is missing a Location: header.',)`

**Comment by shadow-light** [comment]
I figured out the `bq` command has a separate copy of httplib2 which I suspect doesn't have the fix that was added for `gcloud`. I manually went in, removed 308, and command worked.

**Comment by busunkim96** [comment]
@shadow-light Ah I see, good to know. 

@shollyman @steffnay where is the issue tracker for the bq command line tool?

**Comment by almegeddon** [comment]
@shadow-light I too am experiencing the same bq load issue, can you elaborate exactly how you removed 308 for your workaround?

**Comment by shadow-light** [comment]
I needed to run `bq load` only once, so I just hacked the httplib2 package that google-api-python was using. I removed 308 from [this line](https://github.com/httplib2/httplib2/blob/9bf300cdc372938f4237150d5b9b615879eb51a1/python3/httplib2/__init__.py#L168).

You **do not** want to do that to a copy of httplib2 that other packages are using because it's a hack specific to the Google API. A 308 *should* usually be treated as a redirect. Also note that `bq` used a separate copy of httplib2 than `gcloud` on my system, which took me a long time to figure out!

**Comment by almegeddon** [comment]
@shadow-light thanks for the details. I was able to downgrade to cloud sdk version 289 where bq load is working fine for large files.

**Comment by bibimorlet** [comment]
I am experiencing this issue too with the same endpoint: https://developers.google.com/bid-manager/v1.1/queries/createquery.

I have two email addresses (accounts) to access DBM/DisplayVideo360 tool.

I don't know if this can help but:
-with one account, the endpoint works and I don't have any problem.
-with the other account, I can't create queries/reports with the method 'createquery' (for all my advertisers!), and I get the mentioned error:
`httplib2.RedirectMissingLocation: Redirected but response is missing a location: header`

I am currently working on the following environment:

- python 3.7
- google-api-python-client 1.7.3
- httplib2                 0.15.0

And I tried to change the versions, upgrade, downgrade, but nothing changes.

Does anyone have more information about this issue?

**Comment by busunkim96** [comment]
The gcloud SDK team is aware of the BQ CLI issue. If you're seeing this on another API, please open a new issue and we'll take a look.

Thanks!

**Comment by RDelbangoUmbrella** [comment]
Issue is appearing again with this endpoint [](https://developers.google.com/bid-manager/v1.1/queries/createquery)

**Comment by RDelbangoUmbrella** [comment]
Issue is appearing again on this endpoint 

https://developers.google.com/bid-manager/v1.1/queries/createquery

**Comment by joeree** [comment]
I second the fact that the problem is happening again. Below is a traceback for this:
```File "/gaia/.local/lib/python3.6/site-packages/googleapiclient/_helpers.py", line 130, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/gaia/.local/lib/python3.6/site-packages/googleapiclient/http.py", line 846, in execute
    method=str(self.method), body=self.body, headers=self.headers)
  File "/gaia/.local/lib/python3.6/site-packages/googleapiclient/http.py", line 164, in _retry_request
    resp, content = http.request(uri, method, *args, **kwargs)
  File "/gaia/.local/lib/python3.6/site-packages/google_auth_httplib2.py", line 198, in request
    uri, method, body=body, headers=request_headers, **kwargs)
  File "/gaia/.local/lib/python3.6/site-packages/httplib2/__init__.py", line 1976, in request
    cachekey,
  File "/gaia/.local/lib/python3.6/site-packages/httplib2/__init__.py", line 1679, in _request
    content,
httplib2.RedirectMissingLocation: Redirected but the response is missing a Location: header.
```

**Comment by mirahudson** [comment]
Also seeing this exact issue as of this morning when hitting the endpoint at https://developers.google.com/bid-manager/v1.1/queries/createquery
